src/lightning/fabric/strategies/deepspeed.py

```diff
@@ -349,6 +349,11 @@ def setup_optimizer(self, optimizer: Optimizer) -> Optimizer:
         """
         raise NotImplementedError(self._err_msg_joint_setup_required())
 
+    @contextmanager
+    def module_init_context(self) -> Generator[None, None, None]:
+        with super().module_init_context(), self.module_sharded_context():
+            yield
```

Comment:
I added one if `Fabric.sharded_model`, but I don't want to add them in the strategy. Otherwise I need to refactor everything just to avoid the deprecation message being shown when the call comes from Fabric. 

I can add the deprecation message to the docstring though.